src/client/kotlin/tech/thatgravyboat/skycubed/features/info/foraging/ParkInfoOverlay.kt

```diff
@@ -0,0 +1,56 @@
+package tech.thatgravyboat.skycubed.features.info.foraging
+
+import net.minecraft.client.gui.GuiGraphics
+import net.minecraft.client.renderer.RenderType
+import net.minecraft.network.chat.Component
+import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyOnSkyBlock
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyWidget
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidget
+import tech.thatgravyboat.skyblockapi.api.events.info.TabWidgetChangeEvent
+import tech.thatgravyboat.skyblockapi.helpers.McClient
+import tech.thatgravyboat.skyblockapi.utils.regex.component.ComponentRegex
+import tech.thatgravyboat.skyblockapi.utils.regex.component.anyMatch
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.TextColor
+import tech.thatgravyboat.skycubed.SkyCubed
+import tech.thatgravyboat.skycubed.api.displays.Displays
+import tech.thatgravyboat.skycubed.features.info.CommonInfoDisplays
+
+object ParkInfoOverlay {
+    private var rainTime: Component? = null
+    private val rainTimeRegex = ComponentRegex(" Rain: (?<time>.*)")
+
+    @Subscription
+    @OnlyOnSkyBlock
+    @OnlyWidget(TabWidget.AREA)
+    fun onWidgetUpdate(event: TabWidgetChangeEvent) {
+        rainTimeRegex.anyMatch(event.newComponents, "time") { (time) ->
+            rainTime = time
+        }
+    }
+
+    private val rainTimeDisplay
+        get() = Displays.background(
+            CommonInfoDisplays.LEFT_LINE,
+            Displays.padding(
+                3, 1, 2, 2, Displays.row(
+                    Displays.padding(1, Displays.sprite(SkyCubed.id("info/icons/bucket"), 8, 8)),
+                    Displays.text(rainTime ?: Text.of("N/A").withColor(TextColor.RED))
```

Comment:
```suggestion
                    Displays.text { rainTime ?: Text.of("N/A").withColor(TextColor.RED) }
```

do this and then make this property a val and not a getter